Client/Frontend/Browser/TabManager.swift

```diff
@@ -436,22 +436,21 @@ class TabManager: NSObject {
         }
     }
 
-    func isSelectedParentTab(afterRemoving tab: Tab) -> Bool {
-        let viableTabs: [Tab] = tab.isPrivate ? privateTabs : normalTabs
+    // Select the most recently visited tab, IFF it is also the parent tab of the closed tab.
+    func selectParentTab(afterRemoving tab: Tab) -> Bool {
+        let viableTabs = tab.isPrivate ? privateTabs : normalTabs
+        guard let parentTab = tab.parent, parentTab != tab, !viableTabs.isEmpty else { return false }
 
-        if let parentTab = tab.parent,
-            let newTab = viableTabs.reduce(viableTabs.first, { currentBestTab, tab2 in
-                if let tab1 = currentBestTab, let time1 = tab1.lastExecutedTime {
-                    if let time2 = tab2.lastExecutedTime {
-                        return time1 <= time2 ? tab2 : tab1
-                    }
-                    return tab1
-                } else {
-                    return tab2
-                }
-            }), parentTab == newTab, tab !== newTab, newTab.lastExecutedTime != nil {
-            // We select the most recently visited tab, only if it is also the parent tab of the closed tab.
-            _selectedIndex = tabs.index(of: newTab) ?? -1
+        var parentTabIsMostRecentUsed = true
+        for candidate in viableTabs {
+            if let time = candidate.lastExecutedTime, time < (tab.lastExecutedTime ?? 0) {
+                parentTabIsMostRecentUsed = false
+                break
```

Comment:
If I'm not mistaken, the for loop here is not the same as the reduce on the left. Here we'll find the first tab that meets our condition and exit the loop. But what if there is another tab that is more recent later in the list? wouldn't we want that instead?

Reply:
The original code is only making a boolean decision on whether any tab is more recent than the parent (which was difficult to decipher from the syntax).
Hence I added`parentTabIsMostRecentUsed`, and this loop early-exits if that is found to be false. Ideas welcome on how that would be made most clear to the reader.